detectors/aws/ecs/ecs.go

```diff
@@ -125,6 +128,26 @@ func (detector *resourceDetector) Detect(ctx context.Context) (*resource.Resourc
 			}
 		}
 
+		arnParts := strings.Split(taskMetadata.TaskARN, ":")
+		// a valid arn should have at least 6 parts
+		if len(arnParts) < 6 {
+			return empty, errCannotParseTaskArn
+		}
+
+		attributes = append(
+			attributes,
+			semconv.CloudRegion(arnParts[3]),
+			semconv.CloudAccountID(arnParts[4]),
```

Comment:
Is this guaranteed to not change upstream?

Reply:
The day the ARN format of AWS changes, we are all in for a funny couple years.

Reply:
Can we quickly (i.e. non-regex based) validate both of these values to ensure they are correct before adding them here?

Reply:
It seems to me frankly overkill. The only think it would do, is to create a more understandable error in a case that is really unlikely to ever occur: to the best of my knowledge, ARNs historically change almost never, and practically never for established services. Way too much software depends on them.

Reply:
For reference, ECS ARNs changed back in 2020 (https://aws.amazon.com/blogs/compute/migrating-your-amazon-ecs-deployment-to-the-new-arn-and-resource-id-format-2/) and it was a Big Thing for migration periods and all :-)